vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/GenericSetVariableSteps.java

```diff
@@ -0,0 +1,148 @@
+     * Actions performed at this step:
+     * <ul>
+     * <li>Saves the value of attribute with name <i>attributeName</i> to the <i>variableName</i>
+     * </ul>
+     * @param attributeName the name of the attribute (for ex. 'name', 'id')
+     * @param scopes The set (comma separated list of scopes e.g.: STORY, NEXT_BATCHES) of variable's scope<br>
+     * <i>Available scopes:</i>
+     * <ul>
+     * <li><b>STEP</b> - the variable will be available only within the step
+     * <li><b>SCENARIO</b> - the variable will be available only within the scenario
+     * <li><b>STORY</b> - the variable will be available within the whole story
+     * <li><b>NEXT_BATCHES</b> - the variable will be available starting from next batch
+     * </ul>
+     * @param variableName A name under which the value should be saved
+     */
+    @When("I set `$attributeName` attribute value of context element to $scopes variable `$variableName`")
+    public void saveContextElementAttributeValueToVariable(String attributeName, Set<VariableScope> scopes,
+            String variableName)
+    {
+        saveVariableIfContextElementPresent(contextElement -> getAssertedAttributeValue(contextElement, attributeName),
+                scopes, variableName);
+    }
+
+    /**
+     * Extracts the value of <b>attribute</b> of element found by <b>locator</b> and saves it to the <b>variable</b>
+     * with the specified <b>variableName</b>
+     * Actions performed at this step:
+     * <ul>
+     * <li>Saves the value of attribute with name <i>attributeName</i> to the <i>variableName</i>
+     * </ul>
+     * @param attributeName the name of the attribute (for ex. 'name', 'id')
+     * @param locator locator to find an element
+     * @param scopes The set (comma separated list of scopes e.g.: STORY, NEXT_BATCHES) of variable's scope<br>
+     * <i>Available scopes:</i>
+     * <ul>
+     * <li><b>STEP</b> - the variable will be available only within the step
+     * <li><b>SCENARIO</b> - the variable will be available only within the scenario
+     * <li><b>STORY</b> - the variable will be available within the whole story
+     * <li><b>NEXT_BATCHES</b> - the variable will be available starting from next batch
+     * </ul>
+     * @param variableName A name under which the value should be saved
+     */
+    @When("I set `$attributeName` attribute value of element located `$locator` to $scopes variable `$variableName`")
+    public void saveAttributeValueOfElementByLocatorToVariable(String attributeName, Locator locator,
+            Set<VariableScope> scopes, String variableName)
+    {
+        locator.getSearchParameters().setVisibility(Visibility.ALL);
+        baseValidations.assertElementExists("The element to extract the attribute", locator)
+                .map(element -> getAssertedAttributeValue(element, attributeName))
+                .ifPresent(attributeValue -> bddVariableContext.putVariable(scopes, variableName, attributeValue));
```

Comment:
we should init variable with either `value` or `null` (to avoid cases when previously created variable contains old value)
I see that it's not a part of this PR, but it should be fixed